discovery_engine/lib/src/domain/engine/engine.dart

```diff
@@ -23,6 +25,9 @@ import 'package:xayn_discovery_engine/src/domain/models/unique_id.dart'
 
 /// Interface to Discovery Engine core.
 abstract class Engine {
+  /// NOTE: this is to enforce subclasses to have these parameters during instantiation
```

Comment:
We don't need to enforce this. If the mock does not need to work is ok.
The real implementation will have it.